Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Operator/Driver.hs

```diff
@@ -37,16 +38,32 @@ getDriverOperatorFetchHubRequests _merchantShortId _opCity mbFrom mbTo mbStatus
   pure $ API.Types.ProviderPlatform.Operator.Driver.OperationHubReqResp {requests = map castHubRequests reqList}
 
 postDriverOperatorRespondHubRequest :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> API.Types.ProviderPlatform.Operator.Driver.RespondHubRequest -> Environment.Flow APISuccess)
-postDriverOperatorRespondHubRequest _merchantShortId _opCity req = do
+postDriverOperatorRespondHubRequest merchantShortId opCity req = do
   now <- getCurrentTime
   Redis.whenWithLockRedis (opsHubRequestLockKey req.operationHubRequestId) 60 $ do
     opHubReq <- SQOHR.findByPrimaryKey (Kernel.Types.Id.Id req.operationHubRequestId) >>= (fromMaybeM (InternalError "Invalid operation hub request id"))
     unless (opHubReq.requestStatus == PENDING) $ Kernel.Utils.Common.throwError (InvalidRequest "Request already responded")
     when (opHubReq.driverId.getId == req.driverId) $ do
       void $ SQOHR.updateStatusWithDetails (castReqStatusToDomain req.status) (Just req.remarks) (Just now) (Just (Kernel.Types.Id.Id req.operatorId)) (Kernel.Types.Id.Id req.operationHubRequestId)
       when (req.status == API.Types.ProviderPlatform.Operator.Driver.APPROVED && opHubReq.requestType == ONBOARDING_INSPECTION) $ do
-        fork "enable driver after inspection" $
-          void $ postDriverEnable _merchantShortId _opCity (Kernel.Types.Id.Id req.driverId)
+        fork "enable driver after inspection" $ do
+          let personId = Kernel.Types.Id.Id req.driverId
+          merchant <- findMerchantByShortId merchantShortId
+          merchantOpCity <- CQMOC.findByMerchantIdAndCity merchant.id opCity >>= fromMaybeM (MerchantOperatingCityNotFound $ "merchantShortId: " <> merchantShortId.getShortId <> " ,city: " <> show opCity)
+          transporterConfig <- findByMerchantOpCityId merchantOpCity.id Nothing >>= fromMaybeM (TransporterConfigNotFound merchantOpCity.id.getId) -- (Just (DriverId (cast personId)))
+          vehicle <- runInReplica $ QVehicle.findById personId >>= fromMaybeM (VehicleDoesNotExist personId.getId)
+          person <- runInReplica $ QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+          let language = fromMaybe merchantOpCity.language person.language
+          driverDocuments <- SStatus.fetchDriverDocuments personId merchantOpCity transporterConfig language
+          processedVehicleDocuments <- SStatus.fetchProcessedVehicleDocuments personId merchantOpCity transporterConfig language (Just vehicle)
+          vehicleDoc <-
+            find (\doc -> doc.registrationNo == vehicle.registrationNo) processedVehicleDocuments
+              & fromMaybeM (InternalError $ "Vehicle doc not found for driverId " <> personId.getId) -- TODO remove this InternalError
```

Comment:
Don't like this part, where we fetch list of docs, and then trying to find doc for current vehicle
Will think how to refactor